projects/core/directives/dropdown/dropdown-context.directive.ts

```diff
@@ -44,6 +62,53 @@ export class TuiDropdownContextDirective extends TuiRectAccessor {
         this.driver.next(false);
     }
 
+    @HostListener('touchmove.silent', [
+        '$event.touches[0].clientX',
+        '$event.touches[0].clientY',
+    ])
+    onTouchMove(x: number, y: number): void {
+        if (!this.isIOS || this.driver.value) {
+            return;
+        }
+
+        if (
+            Math.hypot(x - this.touchStartRect.x, y - this.touchStartRect.y) >
+            tuiLongTapMoveThreshold
+        ) {
+            this.onTouchEnd();
+        }
+
+        this.currentRect = tuiPointToClientRect(x, y);
```

Comment:
How about we will only keep `currentRect` set at the initial touchstart and not update it on move, then we will not need to keep touchStartRect, so we will only use touchmove listener to cancel if it's moved too far.